server/controllers/complaintHandler.js

```diff
@@ -0,0 +1,219 @@
+const DocComplaint = require('../database/models/docComplaint');
+const UserComplaint = require('../database/models/userComplaint');
+const authentication = require('../authentication');
+
+const createDocComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    let doccomplaintinst = new DocComplaint({
+      text: req.body.text,
+      fromUserId: req.user._id,
+      revisionId: req.body.revisionId,
+      docId: req.body.docId 
+    });
+    
+    doccomplaintinst.save(function (err, doccomp) {
+      if (err) {
+        res.status(500).json({error:"Could not create the Document Complaint."});
+      }
+      else {
+        console.log("Document Complaint saved to DocComplaints collection.");
+        res.status(200).json({msg:"Document Complaint Created."});
+      }
+    });
+  }
+}
+
+const retrieveDocComplaint = (req, res) => {
+  DocComplaint.findOne({_id:req.params.complaintId}, function(err, result){
+    if (err || !result) {
+      res.status(404).json({error:"Could not retrieve the Document Complaint."});
+    }
+    else {
+      res.status(200).json({docComplaint:result});
+    }
+  });
+};
+
+const getDocComplaintList = (req, res) => {
+  DocComplaint.find({}, function(err, results){
+    if (err || !result) {
+      res.status(404).json({error:"Could not retrieve Document Complaints."});
+    }
+    else {
+      res.status(200).json({docComplaintList:results});
+    }
+  });
+};
+
+const deleteDocComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    DocComplaint.deleteOne({_id:req.params.complaintId}, function(err){
+      if (err) {
+        res.status(500).json({error:"Could not delete the Document Complaint."});
+      }
+      else {
+        res.status(200).json({msg:"Document Complaint Deleted."});
+      }
+    });
+  }
+}
+
+const retrieveUnprocessedDocComplaints = (req, res) => {
+  DocComplaint.find({processed: false, docId: req.body.docId}, function(err, results){
+    if (err || !result) {
+      res.status(404).json({error:"Could not retrieve Unprocessed Document Complaints."});
+    }
+    else {
+      res.status(200).json({unprocessedDocComplaints:results});
+    }
+  });
+}
+
+const processDocComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    DocComplaint.updateOne({_id: req.body.docComplaintId}, {processed: true}, function(err, result) {
+      if(err || !result){
```

Comment:
if(err) {
result is not the document that is updated
Fix for process user complaints as welll